File: tests/bench/sieve.py
N = 2000000
def count_primes(n):
    sieve = [True] * (n + 1)
    sieve[0] = False
    sieve[1] = False
    i = 2
    while i * i <= n:
        if sieve[i]:
            j = i * i
            while j <= n:
                sieve[j] = False
                j += i
        i += 1
    return sum(1 for i in range(n + 1) if sieve[i])
print(count_primes(N))

File: tests/bench/sieve.mjs
const N = 2000000;
function countPrimes(n) {
  const sieve = new Array(n + 1).fill(true);
  sieve[0] = false; sieve[1] = false;
  for (let i = 2; i * i <= n; i++) {
    if (sieve[i]) for (let j = i * i; j <= n; j += i) sieve[j] = false;
  }
  let c = 0;
  for (let i = 0; i <= n; i++) if (sieve[i]) c++;
  return c;
}
console.log(countPrimes(N));
